online_test.go

```diff
@@ -400,15 +400,29 @@ func TestProxyProto(t *testing.T) {
 	}
 }
 
-func TestHostname(t *testing.T) {
+var letters = []byte("abcdefghijklmnopqrstuvwxyz")
+
+func randString(n int) string {
+	b := make([]byte, n)
+	for i := range b {
+		b[i] = letters[rand.Intn(len(letters))]
+	}
+	return string(b)
+}
+
+func TestDomain(t *testing.T) {
 	paidTest(t)
 	ctx := context.Background()
 
+	prefix := randString(8)
+	domain := fmt.Sprintf("%s.ngrok.io", prefix)
+	expectedUrl := fmt.Sprintf("https://%s", domain)
+
 	tun, exited := serveHTTP(ctx, t, nil,
-		config.HTTPEndpoint(config.WithDomain("foo.robsonchase.com")),
+		config.HTTPEndpoint(config.WithDomain(domain)),
 		helloHandler,
 	)
-	require.Equal(t, "https://foo.robsonchase.com", tun.URL())
```

Comment:
This probably wouldn't have worked for anyone but me, or someone authenticated with our dev account. Whoops :stuck_out_tongue:

Reply:
and still might fail ;) How does randomly server-selected domain work? Just don't pass `WithDomain` option?

Reply:
Yeah, but this test is specifically to test the option, so falling back to a random one wouldn't really make sense :stuck_out_tongue: